### ISSUE TITLE
StreamReader usage is wrong

### ISSUE THREAD

**Issue by BrannonKing** [nats-io/nats.net.v1#181]
There's a comment in the code about purposely leaking the StreamReaders to keep the parent stream open. This approach is incorrect and is probably causing a memory leak. The StreamReaders need to be disposed. They have a constructor parameter that can be set to leave the source stream open, which should be utilized. The call to create them should look something like this:
```csharp
... new StreamReader(br, Encoding.UTF8, false, 4096, true) ...
```
It should then get disposed after the last ReadLine call (maybe utilizing the `using` statement if it might throw). The Encoding should be set to the real encoding, whatever that is (unless you really want to detect it from byte marks, but it seems that it would always be the same). 4096 should be smaller if your average header data that you're looking for is less than that, etc.

**Comment by ColinSullivan1** [comment]
@BrannonKing , let me take a look.  Thanks!

**Comment by ColinSullivan1** [comment]
@BrannonKing , I'll modify the API to explicitly leave the underlying stream open - always a good idea to explicitly signal intent in code.  Thanks!

That being said, I do not believe this was a memory leak.  If you do not explicitly `Close()` (or `Dispose()`) the stream, the garbage collector will not close the underlying stream.   I didn't see this as an issue in previous testing for leaks, and with current code before the change I didn't find leaks around the stream using the debugger manually and counting object instances.

**Comment by ColinSullivan1** [comment]
Resolved by #183.  Thank you for raising the issue.
